lib/code-caller-python.js

```diff
@@ -169,42 +239,58 @@ class PythonCaller {
 
     this.state = IN_CALL;
     this._checkState();
-    debug(`exit call(), state: ${String(this.state)}, uuid: ${this.uuid}`);
+    this.debug('exit call()');
   }
 
-  /*
-   * @param {function} callback - A callback(err, success) function. If 'success' is false then this PythonCaller should be discarded and a new one created by the parent.
+  /**
+   * Instructs the caller to restart, which means exiting the forked process
+   * and forking a new one from the zygote.
+   *
+   * @param {(err: Error | undefined, success?: boolean) => void} callback
    */
   restart(callback) {
-    debug(`enter restart(), state: ${String(this.state)}, uuid: ${this.uuid}`);
+    this.debug('enter restart()');
     this._checkState([CREATED, WAITING, EXITING, EXITED]);
```

Comment:
@mwest1066 I noticed that we're frequently calling `this._checkState()` but ignoring the return value, which is the only thing that tells us anything useful - I don't *think* this is intentional, but it seems like a pretty big oversight. (This did not change in this PR, it was true on master too.)

Reply:
I guess my thinking here was that the logging of state machine errors was the key outcome of a call to `_checkState()`. I seem to remember that these errors would show up on the console when running PL locally, making it helpful for debugging when writing the code originally. However, I fully agree that we should be taking some more visible action now (especially in production). Is there some simple way to get any `_checkState()` errors to bubble up so that they'll be visible as course issues or something like that?